crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -4102,6 +4146,32 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn comparison_string_literals() -> anyhow::Result<()> {
+        let mut db = setup_db();
+        db.write_dedented(
+            "src/a.py",
+            r#"
+            def str_instance() -> str: ...
+            a = "abc" == "abc"
+            b = "ab_cd" <= "ab_ce"
+            c = "abc" in "ab cd"
+            d = "" not in "hello"
+            e = "--" is "--"
```

Comment:
Let's add a test showing that `is` returns `Literal[False]` with unequal strings, and also both variants of `is not` (equal and unequal strings).